plenum/server/pool_manager.py

```diff
@@ -227,15 +227,16 @@ def onPoolMembershipChange(self, txn):
         self._set_node_order(nodeNym, nodeName)
 
         def _updateNode(txn_data):
-            if {NODE_IP, NODE_PORT, CLIENT_IP, CLIENT_PORT}. \
-                    intersection(set(txn_data[DATA].keys())):
-                self.nodeHaChanged(txn_data)
-            if VERKEY in txn_data:
-                self.nodeKeysChanged(txn_data)
             if SERVICES in txn_data[DATA]:
                 self.nodeServicesChanged(txn_data)
-            if BLS_KEY in txn_data[DATA]:
-                self.node_blskey_changed(txn_data)
+            if txn_data[DATA][ALIAS] in self.node.nodeReg:
+                if {NODE_IP, NODE_PORT, CLIENT_IP, CLIENT_PORT}. \
+                        intersection(set(txn_data[DATA].keys())):
+                    self.nodeHaChanged(txn_data)
+                if VERKEY in txn_data:
+                    self.nodeKeysChanged(txn_data)
+                if BLS_KEY in txn_data[DATA]:
+                    self.node_blskey_changed(txn_data)
```

Comment:
I think there should be a test where a node is promoted with changed HA (for example port) to check that changing of conditions sequence does not break the logic.

Reply:
ok